lib/ASTUtils.cc

```diff
@@ -249,4 +250,24 @@ std::set<const clang::Stmt *> subtrees(const clang::Stmt *ST) {
     }
     return Subtrees;
 }
+
+bool endsWithCompound(const clang::Stmt *ST) {
+    if (!ST) {
+        return false;
+    } else if (clang::isa<clang::CompoundStmt>(ST)) {
+        return true;
+    } else if (auto While = clang::dyn_cast<clang::WhileStmt>(ST)) {
+        return endsWithCompound(While->getBody());
+    } else if (auto For = clang::dyn_cast<clang::ForStmt>(ST)) {
+        return endsWithCompound(For->getBody());
+    } else if (auto If = clang::dyn_cast<clang::IfStmt>(ST)) {
+        if (auto Else = If->getElse()) {
+            return endsWithCompound(Else);
+        } else {
+            return endsWithCompound(If->getThen());
+        }
```

Comment:
I think this may need to handle a switch statement, I'm able to create a case like this:
```c
#define TEST(X) switch(X) {case 1: ;}

int main(int argc, char *argv[])
{

        TEST(1)
        return 0;
}
~```
`DoesBodyEndWithCompoundStmt` will be false in this case, and MerC  translates the macro and causes a compilation error.

Reply:
Ah nice catch, I just added a fix